### DISCUSSION TITLE
Advise for converting ECG waveform to wfdb when data may be missing

### DISCUSSION THREAD

**Discussion by msjoding** [wfdb/wfdb-spec#19] [Q&A]
Hello, we trying to store our waveform data into wfdb format at Michigan and I'm looking for advise on how to best handle situations when data are missing/ECG collection frequency changes. We encountered a situation where a patients ECG signal, which was at 240 hertz, drops for 1 second, then comes back at 90 hertz, then increases to 120 hertz, then finally the ECG signal returns to 240 hertz. When the signal is getting collected at 120 hertz, we can still make out the ECG morphology so its not complete noise. I'm wondering if the group has advise when storing this data in wfdb format?  We could try keeping the data at 240 hertz, but adding NaN values when needed, create segments of the ECG, broken up by when the data changes hertz, or another apporach. Thank you for your advise.

**Comment by briangow** [comment]
@msjoding , I had drafted a response for your question but apparently didn't post it. Sorry for the delay.

Going with your suggestion "create segments of the ECG, broken up by when the data changes hertz" would make sense. Having said that, this situation seems very strange. Do you know why the frequency for a given channel is jumping?
